### PR TITLE
Oracle websocket events

### PR DESCRIPTION
**Pull request by hanssv**
Add subscription mechanism for transaction events, and use it for Oracle
    query/response. Also:
    - Add query to interaction object
    - Update to base58 encoding

### CODE REVIEW

**Comment by hanssv** [comment]
I need to convince Dialyzer that this is actually correct...

**Comment by lucafavatella** [comment]
@hanssv Could please specify which pivotal ticket this PR finishes - for clarifying scope.

**Comment by hanssv** [comment]
This patch finishes #153833523 and #153833554

File: apps/aecore/src/aec_subscribe.erl [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+add_subscribed(Id, Event = {aeo, {query, _}}, Sub = #sub{ oracle_query = OQs }) ->
+    Sub#sub{ oracle_query = [{Id, Event} | OQs] };
+add_subscribed(Id, Event = {aeo, {response, _}}, Sub = #sub{ oracle_response = ORs }) ->
+    Sub#sub{ oracle_response = [{Id, Event} | ORs] };
+add_subscribed(Id, Event, Sub) ->
+    lager:error("Unhandled subscription event kind ~p from ~p", [Event, Id]),
+    Sub.
+
+del_subscribed(Id, Sub = #sub{ oracle_query = OQs, oracle_response = ORs }) ->
+    Sub#sub{ oracle_query    = [ OQ || OQ = {Id1, _} <- OQs, Id =/= Id1 ],
```
Comment (lucafavatella) [inline apps/aecore/src/aec_subscribe.erl]: `{Id1, _} <- OQs` filters, discarding silently items that do not match e.g. after code change from 2-tuple. May I suggest either of the two improvements for maintainability:
* Change this list comprehension to usage of `lists` module with explicit pattern matching;
* Type `#sub.oracle_query` - for Dialyzer.

File: apps/aecore/src/aec_subscribe.erl [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+             oracle_response = [ OR || OR = {Id1, _} <- ORs, Id =/= Id1 ] }.
```
Comment (lucafavatella) [inline apps/aecore/src/aec_subscribe.erl]: Please refer to comment above re `#sub.oracle_query`.

File: apps/aehttp/src/ws_int_dispatch.erl [hunk 3]
```diff
@@ -21,7 +21,8 @@ execute(Target, Action, Payload) ->
 
 -spec is_valid(Target :: atom(), Action :: atom(), Payload :: map()) ->
                 true | {false, ErrMsg :: binary}.
-is_valid(chain, get, #{<<"height">> := Height}) when Height < 0->
+is_valid(chain, get, #{<<"height">> := Height})
+    when not is_integer(Height); Height < 0 ->
```
Comment (lucafavatella) [inline apps/aehttp/src/ws_int_dispatch.erl]: (Not needed for this PR) Usage of `?GENESIS_HEIGHT`?

File: apps/aehttp/src/ws_handler.erl [hunk 4]
```diff
@@ -43,6 +54,12 @@ websocket_info({gproc_ps_event, Event, Data}, Req, State) ->
 websocket_info({send, SenderName, Action, Payload}, Req, State) ->
     Msg = create_message(SenderName, Action, Payload),
     {reply, {text, jsx:encode(Msg)}, Req, State};
+websocket_info({oracle_query_tx, QTx}, Req, State) ->
+    Msg = create_message_from_event(oracle_query_tx, QTx),
+    {reply, {text, jsx:encode(Msg)}, Req, State};
```
Comment (lucafavatella) [inline apps/aehttp/src/ws_handler.erl]: Are the outputs (and inputs) of the WS API documented in any schema? It looks like [Swagger has support for WebSockets (ws/wss) both in version 2 (that we use) and version 3](https://github.com/OAI/OpenAPI-Specification/issues/1140).

File: apps/aeoracle/src/aeo_interaction.erl [hunk 5]
```diff
@@ -104,6 +116,7 @@ serialize(#interaction{} = I) ->
                  , #{<<"sender_address">>  => sender_address(I)}
                  , #{<<"sender_nonce">>    => sender_nonce(I)}
                  , #{<<"oracle_address">>  => oracle_address(I)}
+                 , #{<<"query">>           => query(I)}
```
Comment (lucafavatella) [inline apps/aeoracle/src/aeo_interaction.erl]: So this is other chain break?

Comment (lucafavatella) [reply apps/aeoracle/src/aeo_interaction.erl]: The integration environment already has [`db13`](https://github.com/aeternity/epoch/blob/26d13590fa4e2aba7bc4a9a683a584180cc60324/deployment/ansible/deploy.yml#L42).

File: apps/aehttp/src/ws_handler.erl (hunk 4 above)
Comment (hanssv) [reply apps/aehttp/src/ws_handler.erl]: The consensus (at least with input from Michal and Dimitar) seems to be that swagger is the wrong thing to do. I will make a structured documentation of the five "endpoints"/"actions" for Oracle - for now I'll put them in `aeterninty/protocol`

File: apps/aehttp/src/ws_int_dispatch.erl (hunk 3 above)
Comment (hanssv) [reply apps/aehttp/src/ws_int_dispatch.erl]: Old code, but yes correct.

File: apps/aeoracle/src/aeo_interaction.erl (hunk 5 above)
Comment (hanssv) [reply apps/aeoracle/src/aeo_interaction.erl]: Yes, true. There are most likely no oracle queries in the integration environment, but better safe than sorry.

File: apps/aecore/src/aec_subscribe.erl (hunk 1 above)
Comment (hanssv) [reply apps/aecore/src/aec_subscribe.erl]: Rewritten.

File: apps/aecore/src/aec_subscribe.erl (hunk 2 above)
Comment (hanssv) [reply apps/aecore/src/aec_subscribe.erl]: Rewritten.

File: apps/aeoracle/src/aeo_interaction.erl (hunk 5 above)
Comment (hanssv) [reply apps/aeoracle/src/aeo_interaction.erl]: Bumped to db14

**Comment by hanssv** [comment]
Fixed conflicts, handled Luca's comments and bumped deployment DB path.

**Pull request merged**